malariagen_data/anoph/to_plink.py

```diff
@@ -0,0 +1,129 @@
+from typing import Optional
+
+import allel  # type: ignore
+import numpy as np
+import os
+import bed_reader
+
+from ..util import dask_compress_dataset
+from .snp_data import AnophelesSnpData
+from . import base_params
+from . import plink_params
+from . import pca_params
+from numpydoc_decorator import doc  # type: ignore
+
+
+class PlinkConverter(
+    AnophelesSnpData,
+):
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        # N.B., this class is designed to work cooperatively, and
+        # so it's important that any remaining parameters are passed
+        # to the superclass constructor.
+        super().__init__(**kwargs)
+
+    @doc(
+        summary="""
+            Write Anopheles biallelic SNP data to the Plink binary file format.
+        """,
+        extended_summary="""
+            This function writes biallelic SNPs to the Plink binary file format. It enables
+            subsetting to specific regions (`region`), selecting specific sample sets, or lists of
+            samples, randomly downsampling sites, and specifying filters based on missing data and
+            minimum minor allele count (see the docs for `biallelic_snp_calls` for more information).
+            The `overwrite` parameter, set to true, will enable overwrite of data with the same
+            hash values (e.g. with a different randomly selected number of sites).
+        """,
+        returns=("Binary Plink output files: .bed, .bim, .fam"),
+        notes="""
+            This computation may take some time to run, depending on your computing
+            environment. Unless the `overwrite` parameter is set to `True`, results will be loaded
+            from a previous computation, if performed, and if the `results_cache` parameter was
+            set when instantiating the API client.
```

Comment:
```suggestion
            This computation may take some time to run, depending on your computing
            environment. Unless the `overwrite` parameter is set to `True`, results will be returned
            from a previous computation, if available.
```

Actually results will be re-used regardless of whether the `results_cache` parameter was set. I.e., this function doesn't make use of the `results_cache` machinery, it handles it's own kind of caching.